src/Compilers/VisualBasic/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.vb

```diff
@@ -0,0 +1,646 @@
+        M(x:=1, x:=2, 3)
+                      ~
+                                        </errors>)
+
+            Dim tree = comp.SyntaxTrees.First()
+            Dim model = comp.GetSemanticModel(tree)
+            Dim nodes = tree.GetCompilationUnitRoot().DescendantNodes()
+            Dim invocation = nodes.OfType(Of InvocationExpressionSyntax)().Single()
+            Assert.Equal("M(x:=1, x:=2, 3)", invocation.ToString())
+            Assert.Null(model.GetSymbolInfo(invocation).Symbol)
+        End Sub
+
+        <Fact>
+        Public Sub TestNamedParams3()
+
+            Dim source =
+<compilation>
+    <file name="Program.vb">
+Class C
+    Shared Sub M(x As Integer, ParamArray y() As Integer)
+    End Sub
+    Shared Sub Main()
+        M(y:=1, 2)
+    End Sub
+End Class
+    </file>
+</compilation>
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(source, parseOptions:=latestParseOptions)
+            comp.AssertTheseDiagnostics(<errors>
+BC30587: Named argument cannot match a ParamArray parameter.
+        M(y:=1, 2)
+          ~
+                                        </errors>)
+
+            Dim tree = comp.SyntaxTrees.First()
+            Dim model = comp.GetSemanticModel(tree)
+            Dim nodes = tree.GetCompilationUnitRoot().DescendantNodes()
+            Dim invocation = nodes.OfType(Of InvocationExpressionSyntax)().Single()
+            Assert.Equal("M(y:=1, 2)", invocation.ToString())
+            Assert.Null(model.GetSymbolInfo(invocation).Symbol)
+        End Sub
+
+        <Fact>
+        Public Sub TestNamedParams4()
+
+            Dim source =
+<compilation>
+    <file name="Program.vb">
+Class C
+    Shared Sub M(x As Integer, ParamArray y() As Integer)
+    End Sub
+    Shared Sub Main()
+        M(x:=1, y:=2, 3)
+    End Sub
+End Class
+    </file>
+</compilation>
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(source, parseOptions:=latestParseOptions)
+            comp.AssertTheseDiagnostics(<errors>
+BC30587: Named argument cannot match a ParamArray parameter.
+        M(x:=1, y:=2, 3)
+                ~
+                                        </errors>)
+        End Sub
+
+        <Fact>
+        Public Sub TestNamedInvalidParams()
+
+            Dim source =
+<compilation>
+    <file name="Program.vb">
+class C
+
+    Shared Sub M(ParamArray x() As Integer, y As Integer)
+    End Sub
+    Shared Sub Main()
+        M(x:=1, 2)
+    End Sub
+End Class
+    </file>
+</compilation>
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(source, parseOptions:=latestParseOptions)
+            comp.AssertTheseDiagnostics(<errors>
+BC30192: End of parameter list expected. Cannot define parameters after a paramarray parameter.
+    Shared Sub M(ParamArray x() As Integer, y As Integer)
+                                            ~~~~~~~~~~~~
+BC30311: Value of type 'Integer' cannot be converted to 'Integer()'.
```

Comment:
Scratch that, The first error confused me and I thought there was no second parameter.